stunnel/stunnel.ml

```diff
@@ -190,9 +190,9 @@ let config_file config extended_diagnosis host port =
          ]
        ; (if is_fips then ["fips=yes"] else ["fips=no"])
        ; ( if extended_diagnosis then
-             ["debug=authpriv.7"]
+             ["debug=authpriv.5"]
```

Comment:
Do we not want level 7 when `extended_diagnosis = true`? I'd have thought we should log as much as possible in this case

Reply:
The extended_diagnosis  is enabled by default, and it will fill up disk partition.
I cannot fix it by just  disable the `extended_diagnosis` as it do more things than logging.

Reply:
I believe we don't need the extended logs. However, log rotation should avoid the problem. I'm fine with reducing the log level for Stunnel.

Reply:
btw,  the main process just hardcode to 5, see
https://github.com/xapi-project/xen-api/blob/master/ocaml/xapi/helpers.ml#L1739
Do we need to hard code 5 here to keep consistency?

Reply:
> I cannot fix it by just disable the extended_diagnosis as it do more things than logging

What else is it used for?

> The extended_diagnosis is enabled by default

I think changing the default makes more sense. Could we change this line instead perhaps? https://github.com/xapi-project/xen-api-libs-transitional/blob/c77ad87cb6f34894f32ccf051e3bf0bc7c12ff45/http-svr/xmlrpc_client.ml#L327

We use extended_diagnosis=true for the slave -> master connection. It might be useful to keep verbose logging here

Reply:
extended_diagnosis controls not only the log level, but also something like whether remove some log files,  we may not want to just disable it here, 

also, I do not find a good reason to enable debug log by default.

Reply:
> extended_diagnosis controls not only the log level, but also something like whether remove some log files, we may not want to just disable it here,

where does it cause log files to be removed? I don't see anything related to that in the usages? https://github.com/xapi-project/xen-api-libs-transitional/search?q=extended_diagnosis

Reply:
Ah, I see it now (github search completely missed it..) https://github.com/xapi-project/xen-api-libs-transitional/blob/c77ad87cb6f34894f32ccf051e3bf0bc7c12ff45/stunnel/stunnel.ml#L316

Reply:
I think it'd be good to just not delete that file ever, and change the default. But I think the current solution is also fine

Reply:
@lindig  please help to approve this PR if you are happy with it, thanks.